Doc/library/typing.rst

```diff
@@ -1230,6 +1232,120 @@ These are not used in annotations. They are building blocks for creating generic
+    see this written using :data:`Unpack <Unpack>` instead, as
+    ``Unpack[Ts]``.)
+
+    Type variable tuples must *always* be unpacked. This helps distinguish type
+    variable types from normal type variables::
+
+        x: Ts          # Not valid
+        x: tuple[Ts]   # Not valid
+        x: tuple[*Ts]  # The correct way to to do it
+
+    Type variable tuples can be used in the same contexts as normal type
+    variables. For example, in class definitions, arguments, and return types::
+
+        Shape = TypeVarTuple('Shape')
+        class Array(Generic[*Shape]):
+            def __getitem__(self, key: tuple[*Shape]) -> float: ...
+            def __abs__(self) -> Array[*Shape]: ...
+            def get_shape(self) -> tuple[*Shape]: ...
+
+    Type variable tuples can be happily combined with normal type variables::
+
+        DType = TypeVar('DType')
+
+        class Array(Generic[DType, *Shape]):  # This is fine
+            pass
+
+        class Array2(Generic[*Shape, DType]):  # This would also be fine
+            pass
+
+        float_array_1d: Array[float, Height] = Array()     # Totally fine
+        int_array_2d: Array[int, Height, Width] = Array()  # Yup, fine too
+
+    However, note that at most one type variable tuple may appear in a single
+    list of type arguments or type parameters::
+
+        x: tuple[*Ts, *Ts]                     # Not valid
+        class Array(Generic[*Shape, *Shape]):  # Not valid
+            pass
+
+    Finally, an unpacked type variable tuple can be used as the type annotation
+    of ``*args``::
+
+        def call_soon(
+                callback: Callable[[*Ts], None],
+                *args: *Ts
+        ) -> None:
+            ...
+            callback(*args)
+
+    In contrast to non-unpacked annotations of ``*args`` - e.g. ``*args: int``,
+    which would specify that *all* arguments are ``int`` - ``*args: *Ts``
+    enables reference to the types of the *individual* arguments in ``*args``.
+    Here, this allows us to ensure the types of the ``*args`` passed
+    to ``call_soon`` match the types of the (positional) arguments of
+    ``callback``.
+
+    For more details on type variable tuples, see :pep:`646`.
+
+    .. versionadded:: 3.11
+
+.. data:: Unpack
+
+    A typing operator that conceptually marks an object as having been
+    unpacked. For example, using the unpack operator ``*`` on a
+    :class:`type variable tuple <TypeVarTuple>` internally uses ``Unpack``
+    to mark the type variable tuple as having been unpacked::
+
+        Ts = TypeVarTuple('Ts')
+        tup: tuple[*Ts]
+        # Effectively does:
+        tup: tuple[Unpack[Ts]]
+
+    In fact, ``Unpack`` can be used interchangeably with ``*`` in the context
+    of types. You might see ``Unpack`` being used explicitly in older versions
+    of Python, where ``*`` couldn't be used in certain places::
+
+        from typing_extensions import TypeVarTuple, Unpack
```

Comment:
```suggestion
        from typing import TypeVarTuple, Unpack
```

Reply:
Oh I actually think this is better as it is, since it's an example pointing out how `Unpack` is useful in code that wants to be backwards-compatible with Python <3.11.

Maybe we should be clearer about why we're importing from `typing_extensions` rather than `typing`, though.

Reply:
🤦 oh right. Thanks for pointing that out. And yeah not everyone knows what `typing_extensions` is (I didn't before I took over typing).

Reply:
Maybe this?

```suggestion
        # we want to be compatible with earlier versions of Python in this module,
        # so import TypeVarTuple from the `typing_extensions` backports package
        # instead of importing it from `typing`
        from typing_extensions import TypeVarTuple, Unpack
```

Reply:
Mm, good point. Have committed a slightly more concise version of your suggestion, Alex.